### PR TITLE
CORTX-29608: Disable set api for non string type values in confstore

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
Signed-off-by: Rahul Tripathi <rahul.tripathi@seagate.com>

# Problem Statement
- non str values should not be allowed to set in conf/kv store. #Supporting byte strings

# Design
-  checking for type of input value provided by user. #Supporting byte strings

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Comment by rahul-tripathi-git** [comment]
Add a test case for binary values. Binary values should be supported!

File: py-utils/src/utils/kv_store/kv_store_collection.py [hunk 1]
```diff
@@ -435,11 +435,17 @@ def get(self, key: str, *args, **kwargs) -> str:
             raise KvError(errno.EINVAL, \
                 "Invalid response from consul: %d:%s", index, data)
 
-    def _set(self, key: str, val: str, *args, **kwargs) -> Union[bool, None]:
+    def _set(self, key: str, val: str, *args, force: bool = False, **kwargs) -> Union[bool, None]:
```
Comment (rohit-k-dwivedi) [inline py-utils/src/utils/kv_store/kv_store_collection.py:438]: `def _set(self, key: str, val: str, force: bool = False, *args **kwargs) -> Union[bool, None]:`

putting `*args` at last ensures value for `force` is not consumed by *args when passed as

`Conf.set('index', 'key', 'value', True)`

Comment (rahul-tripathi-git) [reply py-utils/src/utils/kv_store/kv_store_collection.py]: Good catch!. Done.

**Review by rohit-k-dwivedi — APPROVED** [review]

**Review by selvakumaar5496 — APPROVED** [review]

**Review by suryakumar1024 — APPROVED** [review]
LGTM

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**